app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaState.kt

```diff
@@ -27,4 +27,6 @@ data class AgendaState(
 
     val agendaItems: Flow<List<AgendaItem>> = flow { emptyList<AgendaItem>() },
```

Comment:
I find having a flow inside a state quite unintuitive, since it doesn't really feel like state (you can't just check the value of it without observing it). In this case, I'd probably prefer using a `combine()` block to combine your screen state with the flow you get from your repo and just have a normal list in the state class

Reply:
OK, I will refactor to use a normal list in the state class.

Reply:
Im not sure I follow you here.

I tried to use your recommendation of `.combine()` and was successful in making the `_agendaState` hold the flow result as a single list, but when I tried to change the `selectedDayIndex` and use the result as a flow, I was not able to update the `agendaDayFlow`. 

Am I understanding that I need to update the `AgendaItems` statically in the `.combine()` and not use a flow for the `selectedDay`? I have updated the `AgendaViewModel` code to reflect this change, but not sure if this approach is the best. 

## Follow up question
Is this is the correct solution you were referring to?

Reply:
Sorry for the confusion! The main issue I'm talking about here is that you will have no way of accessing the current agenda items without launching a new flow collector (even if it's just done via `.first()`, you'll have another DB query for that). 

What would make sense in my head is a setup like this. Might need a bit of adjustment with savedStateHandle and such, but the idea is that we just flatten the flow from the repo (so we get rid of the Flow<*> wrapper and just get the list emissions which we can use to update the agenda state):
```
private val _selectedDayIndex = MutableStateFlow<Int?>(null) // Can remain private
private val agendaItems = _selectedDayIndex
    .flatMapLatest { index ->
        agendaRepository.getAgendaForDayFlow(
            getDateForSelectedDayIndex(index)
        )
    }
    .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())

private val _agendaState = MutableStateFlow(
    AgendaState(
        errorMessage = errorMessage,
        selectedDayIndex = selectedDayIndex,
))
val agendaState = combine(
    _agendaState,
    agendaItems
) { state, items -> 
    // If performance is super important, you could also check here if the list changed and only then copy it
    state.copy(
        agendaItems = items
    )
}.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), AgendaState())
```

Reply:
OK, i have refactored this.